support/camel-k-maven-plugin/src/main/java/org/apache/camel/k/tooling/maven/GenerateCatalogMojo.java

```diff
@@ -524,23 +525,26 @@ private void addCapabilities(RuntimeSpec.Builder runtimeSpec, CamelCatalogSpec.B
         artifacts.clear();
         artifacts.add(Artifact.from("org.apache.camel.quarkus", "camel-quarkus-management"));
         artifacts.add(Artifact.from("org.apache.camel.quarkus", "camel-quarkus-jaxb"));
-        artifacts.add(Artifact.from("org.jolokia", "jolokia-jvm"));
-        addCapabilityAndDependecies(runtimeSpec, catalogSpec, "jolokia", artifacts, false);
+        artifacts.add(Artifact.from("org.jolokia", "jolokia-agent-jvm", "javaagent"));
+        addCapabilityAndDependecies(runtimeSpec, catalogSpec, "jolokia", artifacts, true);
     }
 
     private void addCapabilityAndDependecies(RuntimeSpec.Builder runtimeSpec, CamelCatalogSpec.Builder catalogSpec, String name,
-        List<Artifact> artifacts, boolean addDependency) {
+            List<Artifact> artifacts, boolean addDependency) {
         if (capabilitiesExclusionList != null && !capabilitiesExclusionList.contains(name)) {
             CamelCapability.Builder capBuilder = new CamelCapability.Builder();
-            artifacts.forEach(artifact -> capBuilder.addDependency(artifact.getGroupId(), artifact.getArtifactId()));
+            artifacts.forEach(artifact -> {
+                capBuilder.addDependency(artifact.getGroupId(), artifact.getArtifactId(), artifact.getClassifier());
+                if (addDependency) {
```

Comment:
Why would this be required?

Reply:
This is for the capability to contains all the required dependencies, it's going to generate the following section in the yaml catalog:
```
spec:
  runtime:
    ...
    capabilities:
      jolokia:
        dependencies:
        - groupId: org.apache.camel.quarkus
          artifactId: camel-quarkus-jaxb
        - groupId: org.apache.camel.quarkus
          artifactId: camel-quarkus-management
        - groupId: org.jolokia
          artifactId: jolokia-agent-jvm
          classifier: javaagent
```

Reply:
I see. What I do not understand is which is the difference with the previous code which I think was doing the same.

Reply:
The previous code retrieved only the first artifact (see index zero) `artifacts.get(0)`.
With the proposed change, I moved the artifacts collection to the capability definition.